app/assets/javascripts/discourse/initializers/live-development.js.es6

```diff
@@ -58,12 +58,16 @@ export default {
           // Refresh if necessary
           document.location.reload(true);
         } else {
-          let themeId = currentThemeId();
-
+          const themeIds = currentThemeIds();
           $("link").each(function() {
             if (me.hasOwnProperty("theme_id") && me.new_href) {
-              let target = $(this).data("target");
-              if (me.theme_id === themeId && target === me.target) {
+              const target = $(this).data("target");
+              const themeId = $(this).data("theme-id");
+              if (
+                themeIds.indexOf(me.theme_id) !== -1 &&
+                target === me.target &&
+                (!themeId || (themeId && themeId === me.theme_id))
```

Comment:
We should be able to shorten this to just `(!themeId || (themeId === me.theme_id))`